socs/agents/hwp_pmx/agent.py

```diff
@@ -256,106 +278,101 @@ def acq(self, session, params):
                  'last_updated': 1649085992.719602}
 
         """
-        sleep_time = 1 / self.f_sample - 0.01
+        PMX = None
+        session.set_status('running')
 
-        with self.lock.acquire_timeout(0, job='acq') as acquired:
-            if not acquired:
-                self.log.warn("Could not start acq because {} is already running"
-                              .format(self.lock.job))
-                return False, "Could not acquire lock."
-
-            session.set_status('running')
-            self.take_data = True
-
-            last_release = time.time()
-            while self.take_data:
-                current_time = time.time()
-                if current_time - last_release > 1.:
-                    last_release = time.time()
-                    if not self.lock.release_and_acquire(timeout=10):
-                        self.log.warn(f"Failed to re-acquire lock, currently held by {self.lock.job}.")
-                        continue
-
-                data = {
-                    'timestamp': current_time,
-                    'block_name': 'hwppmx',
-                    'data': {}
-                }
+        self._clear_queue()
```

Comment:
@jlashner This needs to be put in the reactor too ... I suggest blockingCallFromThread.